Made OptionRow and OptionButton the same height, and fixed Night Mode being squashed to the top

agzuniverse opened the pull request:
Added a divider above the "Night Mode" OptionRow in settingsCard. Also added padding to the Label component in OptionRow, so it matched the styling of Label component in OptionButton, which made both of them the same height.

Before:
![screenshot_20180330-022710](https://user-images.githubusercontent.com/22353313/38130144-75a00872-341f-11e8-99c1-bcc77350c2bb.png)

After:
![screenshot_20180330-022612](https://user-images.githubusercontent.com/22353313/38130148-7d4b1012-341f-11e8-9f6f-3baea515fcbc.png)


Also updated the developer guide, added the missing step to run `react-native start` before `react-native run-android` and a command to increase inotify watches to fix a common error while running `react-native start`.

UPDATE:
Screenshot before the OptionRow component was changed, notice the difference in height between the OptionRow and the OptionButton components beneath it.
![screenshot_20180401-173252](https://user-images.githubusercontent.com/22353313/38173249-a1852f16-35d8-11e8-8f07-9937d35770dd.png)

borisyankov commented on docs/developer-guide.md:
```diff
@@ -56,9 +56,11 @@ for starting out, just get either one working so you can play with the app.
 * When running on a physical device, if the device has Zulip installed
   from the Play Store, you may need to uninstall that version first.
 * Commands once you've set up:
+  * `react-native start` - start the packager which builds the JavaScript file
```
I think this is potentially helpful, but you need to describe to the user why would they prefer that over `npm run ios-min` or the other alternatives.

gnprice replied:
Huh -- in fact I've never used this command. Just `react-native run-android` works for me, and I do get the packager running.

So I have no idea why this would be helpful. Would be happy to learn, though. :)

borisyankov commented on docs/developer-guide.md:
```diff
@@ -56,9 +56,11 @@ for starting out, just get either one working so you can play with the app.
   * `react-native run-android` - build and run on an active emulator
     or USB-connected device
   * `npm run android-min` or `npm run android-max` - runs in an Android emulator, emulator has to be run manually before this command
+If `react-native start` gives you an error, try tunning `echo fs.inotify.max_user_watches=524288 | sudo tee -a /etc/sysctl.conf && sudo sysctl -p` first to increase the number of inotify watches.
```
Interesting. When does this happen? Is it OKS dependent?

gnprice commented:
Thanks @agzuniverse !

Quick GitHub workflow tip: it looks like this is a revised version of #2168. It's best to update the existing pull request in that case (by pushing to your branch that you created the PR from), rather than make a new one -- that way the discussion stays connected.

gnprice commented:
I've merged the divider commit, and I rewrote the explanation of the inotify issue and committed that too. Thanks again!

For the padding change: the explanation in this new PR message is helpful, but as I said on #2173, I couldn't spot a difference in the actual UI in the app. Would you
* post in this PR a pair of before/after screenshots of that change alone (with the divider already in the "before"), and
* add your explanation to the actual commit message? (Not the same thing as the PR message -- use `git commit -a` or `git rebase -i` to edit the actual commits.)

agzuniverse commented:
@gnprice I have updated the PR with a screenshot showing the OptionRow before I changed it, but with the divider in place to show the UI change. I have also changed the commit message to be more descriptive.

borisyankov commented:
Thanks @agzuniverse Looks good.

gnprice commented:
Thanks @agzuniverse !

One quick bit of Git commit style for the future, which is important for us reading through history: please keep the summary of the commit message (the first line) to 60-70 columns at most. Additional details are great to put in paragraphs in the body of the message.

This is also a good time to take a look through our [commit style guide](https://zulip.readthedocs.io/en/latest/contributing/version-control.html) -- there are more points there which you might find helpful in the future.
